File: win-prog/ProgWinCS/Paths Regions and Clipping/LineArcCombo/LineArcCombo.cs
//-------------------------------------------
// LineArcCombo.cs © 2001 by Charles Petzold
//-------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class LineArcCombo: PrintableForm
{
     public new static void Main()
     {
          Application.Run(new LineArcCombo());
     }
     public LineArcCombo()
     {
          Text = "Line & Arc Combo";
     }
     protected override void DoPage(Graphics grfx, Color clr, int cx, int cy)
     {
          Pen pen = new Pen(clr, 25);

          grfx.DrawLine(pen,  25, 100, 125, 100);
          grfx.DrawArc (pen, 125,  50, 100, 100, -180, 180);
          grfx.DrawLine(pen, 225, 100, 325, 100);
     }
}

File: win-prog/ProgWinVB/Paths Regions and Clipping/LineArcCombo/LineArcCombo.vb
'---------------------------------------------
' LineArcCombo.vb (c) 2002 by Charles Petzold
'---------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class LineArcCombo
    Inherits PrintableForm

    Shared Shadows Sub Main()
        Application.Run(New LineArcCombo())
    End Sub

    Sub New()
        Text = "Line & Arc Combo"
    End Sub

    Protected Overrides Sub DoPage(ByVal grfx As Graphics, _
            ByVal clr As Color, ByVal cx As Integer, ByVal cy As Integer)
        Dim pn As New Pen(clr, 25)
        grfx.DrawLine(pn, 25, 100, 125, 100)
        grfx.DrawArc(pn, 125, 50, 100, 100, -180, 180)
        grfx.DrawLine(pn, 225, 100, 325, 100)
    End Sub
End Class
